addons/website_sale_stock/models/website.py

```diff
@@ -1,15 +1,25 @@
 # -*- coding: utf-8 -*-
-from odoo import api, models
+from odoo import api, fields, models
+
 
 class Website(models.Model):
     _inherit = 'website'
 
+    @api.model
+    def _default_warehouse_id(self):
+        company = self.env.user.company_id.id
```

Comment:
we should probably uses self.company_id no?
what you think about?

Reply:
is it set before default?

Reply:
yes, you have right.
So dont make a default value.

btw, the code will fallback to the same logic in _prepare_sale_order_values if not default value is set.

I propose to just add domain on res config setting to restrict the warehouse to the current company.